### DISCUSSION TITLE
how to contribute

### DISCUSSION THREAD

**Discussion by NAVJOT-786** [kunjgit/GameZone#3812] [Ideas]
Creating a pull request (PR) involves several steps from cloning a repository to submitting the PR. Here's a detailed guide for a typical workflow using Git and GitHub:

**1. Fork the Repository**

Go to the repository on GitHub you want to contribute to.
Click the "Fork" button at the top-right corner. This creates a copy of the repository under your GitHub account.

**3. Clone Your Forked Repository**
Open a terminal or Git Bash.
Clone the forked repository to your local machine using the following command:

`git clone https://github.com/your-username/repository-name.git`

Navigate into the cloned repository:

`cd repository-name`

**4. Add the Original Repository as a Remote**
Add the original repository as an upstream remote to keep your fork in sync with the main repository:

`git remote add upstream https://github.com/original-owner/repository-name.git`

**5. Create a New Branch**
Create a new branch for your changes:

`git checkout -b your-branch-name`

**6. Make Your Changes**

Make the necessary changes to the codebase.
Stage the changes:

`git add .`

Commit the changes with a meaningful commit message:

`git commit -m "Your detailed commit message"`

**7. Push the Changes to Your Fork**
Push your changes to your forked repository:

`git push origin your-branch-name`

**8. Create a Pull Request on GitHub**
Go to your forked repository on GitHub.
You should see a notification suggesting to create a pull request for the recently pushed branch.
Click "Compare & pull request".
Fill out the PR form:
Ensure the base repository is the original repository and the base branch is the branch you want to merge into (usually main or master).
Provide a clear and concise title and description for your PR.
Click "Create pull request".

**9. Keep Your Fork Updated (Optional but Recommended)**
Fetch the latest changes from the original repository:

`git fetch upstream`
Merge the changes into your local main branch:

```
git checkout main
git merge upstream/main
```
Push the updated main branch to your fork:

`git push origin main`

**Additional Tips**
**Review and Address Feedback:** Once your PR is submitted, the repository maintainers might provide feedback or request changes. Be responsive and make the necessary updates.
**Run Tests:** Ensure all tests pass before submitting your PR. This can often be done using commands like npm test or pytest, depending on the project's setup.
F**ollow Contribution Guidelines:** Some projects have specific contribution guidelines or a code of conduct. Ensure you read and follow these to increase the chances of your PR being accepted.

**Comment by Lincoln-Laishram** [comment]
hey i want to contribute a game of mine i just created recently as my first open source contribution. 
Game url: https://space-game-one.vercel.app/
It is only playable in large screen like desktop and PC for now
I haven't add touch features but ill add it and also make it device responsive. 
If you find this game interesting i want to contribute my game
Ill wait for your approval
